lib/LTP/utils.pm

```diff
@@ -273,24 +274,39 @@ sub schedule_tests {
         loadtest_kernel 'ltp_init_lvm';
     }
 
-    parse_runfiles($cmd_file, $test_result_export);
+    parse_runfiles($cmd_file, $test_result_export, $suffix);
 
     if (check_var('KGRAFT', 1) && check_var('UNINSTALL_INCIDENT', 1)) {
         loadtest_kernel 'uninstall_incident';
-        parse_runfiles($cmd_file, $test_result_export, '_postun');
+        parse_runfiles($cmd_file, $test_result_export, $suffix . '_postun');
     }
 
-    shutdown_ltp(run_args => testinfo($test_result_export));
+    shutdown_ltp(run_args => testinfo($test_result_export))
+      unless get_var('LIBC_LIVEPATCH');
 }
 
 sub parse_openposix_runfile {
     my ($name, $cmds, $cmd_pattern, $cmd_exclude, $test_result_export, $suffix) = @_;
+    my $ulp_test = get_var('LIBC_LIVEPATCH', 0);
     my $whitelist = LTP::WhiteList->new();
 
+    if ($ulp_test) {
+        assert_script_run('export LD_PRELOAD=/usr/lib64/libpulp.so.0');
```

Comment:
this seems it can be one liner

Reply:
Fixed. Originally there was one more debug command and I forgot to reformat the condition after removing it.